### ISSUE TITLE
Validation against some patterns fails incorrectly

### ISSUE THREAD

**Issue by duckontheweb** [Stranger6667/jsonschema-rs#180]
First of all, thanks for this crate; it's been hugely useful to me as I'm learning Rust.

It seems like regex matching on a `"pattern"` for a `"string"` type is not working correctly with some special characters. Checking the string `"CC-BY-4.0"` against the pattern `"^[\\w\\-\\.\\+]+$"` should validate successfully (as demonstrated 
in the online [JSON Schema Validator example](https://www.jsonschemavalidator.net/s/yvxSWxy3)), but instead fails with this error:

```
Validation error: '"CC-BY-4.0"' does not match '^[\w\-\.\+]+$'
```

[Here is a full example](https://replit.com/@JonDuckworth1/JSONSchemaPatterns) of the failed validation on Replit.

**Comment by Stranger6667** [comment]
Thank you so much for such a great report! I am glad that this crate is useful for you :)

You are indeed right. I completely missed that `\w` in ECMA 262 is equivalent to `[A-Za-z0-9_]`, but the current implementation goes only for `[A-Za-z]` (similarly, it is done for `\W`). The initial goal was to exclude additional punctuation (like `︳`) that the `regex` crate matches on `\w` and ECMA 262 does not. 

As I see, the current approach for translation still doesn't cover cases like `\\w`, which will be translated to `\[A-Za-z]`. I'll create a separate issue for that.

**Comment by Stranger6667** [comment]
I'll fix it tomorrow :)

**Comment by duckontheweb** [comment]
> As I see, the current approach for translation still doesn't cover cases like `\\w`, which will be translated to `\[A-Za-z]`. I'll create a separate issue for that.

Based on the error that got reported it seemed like this was being handled correctly, but it's possible that it isn't consuming that escape character until it prints the error message.

**Comment by Stranger6667** [comment]
I mean specifically the `r"\\w"` raw string which corresponds to `"\\\\w"` regular string. 

Based on your example, this should not fail:

```
extern crate jsonschema;

use jsonschema::{Draft, JSONSchema, CompilationError};
use serde_json::json;

fn main() -> Result<(), CompilationError> {
    let schema = json!({        
      "type": "string",
      "pattern": "\\\\w"
    });
    let instance = json!("\\w");
    let compiled = JSONSchema::options()
        .with_draft(Draft::Draft7)
        .compile(&schema)?;
    let result = compiled.validate(&instance);
    assert!(result.is_ok());
    Ok(())
}
```

**Comment by Stranger6667** [comment]
Both fixes are released in `0.6.1`
